common/courses/states.ts

```diff
@@ -152,6 +162,20 @@ export async function cancelSubcourse(user: User, subcourse: Subcourse) {
     }
     await sendSubcourseCancelNotifications(course, subcourse);
     logger.info(`Subcourse (${subcourse.id}) was cancelled`);
+
+    const courseInstructors = await prisma.subcourse_instructors_student.findMany({ where: { subcourseId: subcourse.id }, select: { student: true } });
+    await prisma.user_achievement.deleteMany({
+        where: {
+            userId: {
+                in: courseInstructors.map((instructor) => {
+                    const user = userForStudent(instructor.student);
+                    return user.userID;
+                }),
+            },
+            group: 'student_offer_course',
```

Comment:
Do we need to use the group selector? Maybe the relation is enough?

Reply:
good point. We could delete all achievements related to that subcourse.